src/helpers/data_normalization.ts

```diff
@@ -45,3 +59,29 @@ export function* iterateItemIdsPositions(sheetId: UID, itemIdsByZones: Record<st
     }
   }
 }
+
+function getCanonicalRepresentation(item: any): string {
+  if (item === null) return "null";
+  if (item === undefined) return "undefined";
+  if (typeof item !== "object") return String(item);
+
+  if (Array.isArray(item)) {
+    const len = item.length;
+    let result = "[";
+    for (let i = 0; i < len; i++) {
+      if (i > 0) result += ",";
+      result += getCanonicalRepresentation(item[i]);
+    }
+    return result + "]";
+  }
```

Comment:
Maybe we can do something like `item.map(getCanonicalRepresentation).join(',')`, but I'm not sure about the performance

Reply:
(as you want)